packages/ui/src/IconAdornment/index.tsx

```diff
@@ -0,0 +1,108 @@
+import { LucideIcon } from 'lucide-react';
+import cn from 'clsx';
+import { Icon, IconSize } from '../Icon';
+
+export type IconAdornmentSize = 'sm' | 'xs';
+
+export interface IconAdornmentProps {
+  /**
+   * The icon import from `lucide-react` to render.
+   *
+   * ```tsx
+   * import { Shield } from 'lucide-react';
+   * <IconAdornment icon={Shield} size="sm" />
+   * ```
+   */
+  icon: LucideIcon;
+  /**
+   * Size of the icon adornment:
+   * - `sm`: 24x24px container with 16px icon
+   * - `xs`: 12x12px container with 8px icon
+   */
+  size: IconAdornmentSize;
+  /**
+   * Whether the icon adornment is disabled
+   */
+  disabled?: boolean;
+  /**
+   * Click handler for the icon adornment
+   */
+  onClick?: () => void;
+  /**
+   * Additional CSS classes
+   */
+  className?: string;
+}
+
+const getSizeClasses = (size: IconAdornmentSize): string => {
+  switch (size) {
+    case 'sm':
+      return cn('size-6 p-1'); // 24x24 with 4px padding
+    case 'xs':
+      return cn('size-3 p-0.5'); // 12x12 with 2px padding
+  }
+};
+
+const getIconSize = (size: IconAdornmentSize): IconSize => {
+  switch (size) {
+    case 'sm':
+      return 'sm'; // 16px
+    case 'xs':
+      return 'xs'; // 10px (closest to 8px)
```

Comment:
issue (ui): according to designs I've found, the icon size of XS should be 12x12

<img width="687" height="664" alt="image" src="https://github.com/user-attachments/assets/d9542e8b-1a0a-4f47-851f-fd27ee1ff31e" />